src/data/selectors/filters.js

```diff
@@ -0,0 +1,35 @@
+const getFilters = state => state.filters || {};
+
+const selectableAssignments = (state) => {
+  const selectedAssignmentType = getFilters(state).assignmentType;
+  const needToFilter = selectedAssignmentType && selectedAssignmentType !== 'All';
+  const exampleResultRow = state.grades.results[0] || {};
+  const allAssignments = exampleResultRow.section_breakdown || [];
```

Comment:
I noticed that the assignment filter disappears from the gradebook when selecting a student group with no users in it but the assignment type filter does not. I understand that the reason that assignment types is still displayed is that the list of assignment types in the course relies on a different endpoint, so we're able to get that data independent of whether or not we have an `exampleResultRow` from the data returned by the grades endpoint, which we use to populate the assignment filter. If there is are no grades data associated with the filters, then I see why assignments would be empty. However, I wonder if might be a better experience to either hide both the assignment type and the assignment drop downs or to persist them and make them disabled in this case. I can imagine a course team member being confused why one drop down disappeared and not the other.

Reply:
Switched to a disabled state. Their size is different so it doesn't really stop the flicker-fest, but I agree this is a bit better.